src/share/classes/com/sun/javatest/regtest/exec/BuildAction.java

```diff
@@ -301,6 +303,19 @@ private Status compileFiles(LibLocn libLocn, boolean isMulti, String moduleName,
         if (implicitOpt != null)
             compArgs.add(implicitOpt);
 
+        // read and add library-specific compilation properties
+        if (libLocn.isLibrary()) {
+            try {
+                var properties = LibraryProperties.of(libLocn);
```

Comment:
The `Locations` class would be a good place to cache the `LibraryProperties` object for a `LibLocn`